src/components/input/inputNumber/inputNumber.tsx

```diff
@@ -0,0 +1,204 @@
+import { InputContainer, type IInputComponentProps } from '../inputContainer';
+
+export interface IInputNumberProps extends Omit<IInputComponentProps, 'onChange'> {
+    /**
+     * Optional string appended to the input value.
+     */
+    suffix?: string;
+    /**
+     * @see IUseNumberMaskProps['onChange']
+     */
+    onChange?: IUseNumberMaskProps['onChange'];
+}
+
+export const InputNumber: React.FC<IInputNumberProps> = (props) => {
+    const { suffix, onChange, ...otherProps } = props;
+    const { containerProps, inputProps } = useInputProps(otherProps);
+
+    const { className: containerClassName, isDisabled, ...otherContainerProps } = containerProps;
+    const {
+        max: inputMax,
+        min: inputMin,
+        step: inputStep,
+        value: inputValue,
+        onBlur,
+        onFocus,
+        onKeyDown,
+        className: inputClassName,
+        ...otherInputProps
+    } = inputProps;
+
+    const { step, min, max, value } = parseInputs(inputValue, inputStep, inputMin, inputMax);
+
+    const [isFocused, setIsFocused] = useState(false);
+    const {
+        ref: numberMaskRef,
+        value: maskedValue,
+        unmaskedValue,
+        setValue,
+    } = useNumberMask({
+        min,
+        max,
+        value,
+        onChange,
+    });
+
+    const suffixedValue = maskedValue && suffix ? maskedValue + suffix : maskedValue;
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (e.key === 'ArrowUp') {
+            handleIncrement();
+        } else if (e.key === 'ArrowDown') {
+            handleDecrement();
+        }
+
+        onKeyDown?.(e);
+    };
+
+    const handleFocus = (e: React.FocusEvent<HTMLInputElement>) => {
+        setIsFocused(true);
+        onFocus?.(e);
+    };
+
+    const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+        setIsFocused(false);
+        onBlur?.(e);
+    };
+
+    const handleIncrement = () => {
+        const parsedValue = Number(unmaskedValue ?? 0);
+
+        // return the input value if it's bigger than the max
+        if (parsedValue > max) {
+            setValue(parsedValue.toString());
+            return;
+        }
+
+        // increment directly to the minimum if value is less than the minimum
+        if (parsedValue < min) {
+            setValue(min.toString());
+            return;
+        }
+
+        // round down to the nearest multiple of the step when the value
+        // is not already a multiple of the step
+        let newValue = parsedValue + step;
+        if (parsedValue % step !== 0) {
+            newValue = Math.floor(parsedValue / step) * step;
+        }
+
+        // increment value with step if it's smaller than the initial value
+        if (newValue < parsedValue) {
+            newValue += step;
+        }
+
+        // ensure the new value is within the min and max range
+        newValue = Math.max(min, Math.min(max, newValue));
+        setValue(newValue.toString());
```

Comment:
I would probably simplify this as:
```suggestion
        // Ensure newValue is a multiple of step value
        const newValue = (Math.floor(parsedValue / step) + 1) * step;

        // Ensure newValue is not greater than max value
        setValue(Math.min(max, newValue).toString());
```